### ISSUE TITLE
Dispatch refuses to use rbenv Ruby

### ISSUE THREAD

**Issue by brandoncc** [tpope/vim-dispatch#239]
I'm experiencing the exact issue as #1 and #9, except with rbenv. I am using a fresh setup of VimBox on MacVim, and iterm2 v3.1.6.

I ran:

```
:Dispatch echo `which ruby`
```

and the result is /usr/local/bin/ruby.

When I run:

```
:Dispatch echo $PATH
```

The result is:

```
/usr/local/opt/imagemagick@6/bin:/usr/local/opt/mysql@5.5/bin:/usr/local/heroku/bin:/Users/brandoncc/.composer/vendor/bin:/Users/brandoncc/.cask/bin:/Users/brandoncc/.rbenv/bin:/usr/local/opt/qt@5.5/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Users/brandoncc/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/usr/local/bin:/tools:/platform-tools
```

I also tried adding `eval "$(rbenv init -)"` to `~/.zshenv` to no avail.

I'm hoping you will have a silver bullet that makes this work. Thanks in advance.

**Comment by tpope** [comment]
What I do in my personal config is add both the `bin` and `shims` dir to `$PATH` in `.zprofile`, and then add `eval "$(rbenv init -|grep -v PATH)"` to `.zshrc`. I find this makes the most sense because you want the `PATH` changes everywhere, while everything else in `rbenv init -` really only makes sense in interactive shells.

**Comment by brandoncc** [comment]
Thank you for the response, Tim. It turns out I had a few issues going on:

1. I don't think I killed iterm between tests, and this seems to have mattered
2. iterm has fish set as the default

As soon as I made the changes you recommended and set iterm to "login shell", everything started working fine. This has the unfortunate downside of me no longer being able to use fish by default. Is there a way to get dispatch.vim to work with fish?

Thanks again.
